### PR TITLE
Check &  re-order + reduce battery sensors

### PR DESCRIPTION
**Pull request by fitool**
For the two sensor groups below; I would rename them as follow to make in more clear and shorter (using 'power' to refer to 'active power') but not sure if this has an impact on existing naming in HA?
SENSOR_METERING_ACTIVE_POWER_L1/2/3 --> SENSOR_METERING_POWER_FEEDIN_L1/2/3
SENSOR_METERING_ACTIVE_POWER_CONSUMED_L1/2/3 --> SENSOR_METERING_POWER_DRAWN_L1/2/3

On my Sunny Boy 5.0 PV inverter, I get a response of 60 times "{"val":null}" for the keys corresponding to the sensors below. I get "null" as I don't have optimizers, but I guess there is one value per PV panel (with a maximum of 60) --> if this is correct, it would mean the sensors only give a value for the first panel?
SENSOR_OPTIMIZER_POWER 
SENSOR_OPTIMIZER_CURRENT
SENSOR_OPTIMIZER_VOLTAGE
SENSOR_OPTIMIZER_TEMP

### CODE REVIEW

**Comment by rklomp** [comment]
Thanks! I see you changed some comment headers. For the comments to group the sensors I use the exact naming as shown in the GUI. Makes it easy to trace back where it is coming from. Are the names the same in the battery GUI or are there differences?

image

File: pysma/const.py [hunk 1]
```diff
@@ -107,20 +139,91 @@
...
+SENSOR_POWER_L2 = {
+    SKEY: "6100_40464100",
+    SNAME: "power_l2",
+    SUNIT: "W",
+}
+SENSOR_POWER_L3 = {
+    SKEY: "6100_40464200",
+    SNAME: "power_l3",
+    SUNIT: "W",
+}
```
Comment (rklomp) [inline pysma/const.py]: In my GUI the `SENSOR_GRID_POWER` and `SENSOR_FREQUENCY` are located under
`# AC Side - Grid measurements`

The sensors `SENSOR_POWER_L1`, `SENSOR_POWER_L2` and `SENSOR_POWER_L3` are located under

`# AC Side - Grid measurements - Active power`

```suggestion
# AC Side - Grid measurements
SENSOR_GRID_POWER = {
    SKEY: "6100_40263F00",
    SNAME: "grid_power",
    SUNIT: "W",
}
SENSOR_FREQUENCY = {
    SKEY: "6100_00465700",
    SNAME: "frequency",
    SUNIT: "Hz",
    SFACTOR: 100,
}

# AC Side - Grid measurements - Active power
SENSOR_POWER_L1 = {
    SKEY: "6100_40464000",
    SNAME: "power_l1",
    SUNIT: "W",
}
SENSOR_POWER_L2 = {
    SKEY: "6100_40464100",
    SNAME: "power_l2",
    SUNIT: "W",
}
SENSOR_POWER_L3 = {
    SKEY: "6100_40464200",
    SNAME: "power_l3",
    SUNIT: "W",
}
```

**Comment by fitool** [comment]
Aha ok, I tried to make the header comments more clear but didn't know it was based on the SMA GUI; I also reordered the sensors to be under the renamed headers (e.g.both total power and power per phase under same header)

Ok for me to keep the initial headers and ordering headers to be consistent with SMA GUI. SBS 3.7 GUI is structured in the same way (some small differences ; e.g. Grid measurements has only subsection 'Active power'):

image

**Comment by rklomp** [comment]
> On my Sunny Boy 5.0 PV inverter, I get a response of 60 times "{"val":null}" for the keys corresponding to the sensors below. I get "null" as I don't have optimizers, but I guess there is one value per PV panel (with a maximum of 60) --> if this is correct, it would mean the sensors only give a value for the first panel?
> SENSOR_OPTIMIZER_POWER
> SENSOR_OPTIMIZER_CURRENT
> SENSOR_OPTIMIZER_VOLTAGE
> SENSOR_OPTIMIZER_TEMP

Theses sensors are generated with the correct idx when optimizers are detected. 
https://github.com/rklomp/pysma/blob/dev/pysma/__init__.py#L433-L449

**Comment by fitool** [comment]
allright, great!

File: pysma/const.py [hunk 2]
```diff
@@ -120,7 +152,86 @@
...
+
+# AC Side - Grid measurements - frequency
+SENSOR_FREQUENCY = {
+    SKEY: "6100_00465700",
+    SNAME: "frequency",
+    SUNIT: "Hz",
+    SFACTOR: 100,
+}
+
```
Comment (rklomp) [inline pysma/const.py]: `SENSOR_FREQUENCY` is now duplicate

```suggestion
```

**Comment by rklomp** [comment]
> For the two sensor groups below; I would rename them as follow to make in more clear and shorter (using 'power' to refer to 'active power') but not sure if this has an impact on existing naming in HA?
> SENSOR_METERING_ACTIVE_POWER_L1/2/3 --> SENSOR_METERING_POWER_FEEDIN_L1/2/3
> SENSOR_METERING_ACTIVE_POWER_CONSUMED_L1/2/3 --> SENSOR_METERING_POWER_DRAWN_L1/2/3
> 

These names where based in the headers from the GUI.
image


From SBFSpot
```
11791=Metering.GridMs.W.phsA\4646912\Power grid feeding L1
11792=Metering.GridMs.W.phsB\4647168\Power grid feeding L2
11793=Metering.GridMs.W.phsC\4647424\Power grid feeding L3
11794=Metering.GridMs.WIn.phsA\4647680\Power drawn from grid phase L1
11795=Metering.GridMs.WIn.phsB\4647936\Power drawn from grid phase L2
11796=Metering.GridMs.WIn.phsC\4648192\Power drawn from grid phase L3
```

I think both names would cover it. Changing it would not have an impact at the moment as these are new sensors that were not present in `pysma` yet.

**Comment by rklomp** [comment]
I think if looks pretty good like this! Could we section the battery sensors a bit? How are they grouped in the GUI?

**Comment by fitool** [comment]
I think 'feed-in' and 'drawn' are more descriptive for what these sensors represent, but keeping consistency in the naming with GUI is also also an argument; maybe keep the GUI naming and add the SBFspot name as in-line comment just to clarify ?

Below print screen of battery section:

image

**Comment by rklomp** [comment]
> I think 'feed-in' and 'drawn' are more descriptive for what these sensors represent, but keeping consistency in the naming with GUI is also also an argument; maybe keep the GUI naming and add the SBFspot name as in-line comment just to clarify ?

The current section comments contain 'feed-in' and 'drawn', so i guess that is clear enough?


> Below print screen of battery section:

We could add some section comments based on this to make it a bit more structured.

**Comment by fitool** [comment]
done!

**Pull request merged**